### PR TITLE
Pcm 1819

### PR DESCRIPTION
**Pull request by jensengar**

### CODE REVIEW

File: test/test-pages/gcba/platform-client.ts [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+    const environment = getInputValue('environment');
+    const clientId = (window as any).environments[environment].clientId;
+    initClientEnvironment(environment);
+
+    client.loginImplicitGrant(clientId, window.location.href)
+      .then(() => {
+        const authInfo = JSON.parse(window.localStorage.getItem(`${persitentName}_auth_data`));
+        (client as any).authentications['PureCloud Auth'].accessToken = authInfo.accessToken;
+
+        (window as any).conversationsAPI = new (window as any).platformClientConversationsApi();
```
Comment (davidmorton-gen) [inline test/test-pages/gcba/platform-client.ts:45]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9580329)
```suggestion
        (window as unknown).conversationsAPI = new (window as any).platformClientConversationsApi();
```

File: test/test-pages/gcba/sdk-controller.ts [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+
+function error (error: SdkError) {
+  let output = `${_getLogHeader('error')}
+    error: ${error.message}\n  details: ${error.details}`;
+
+  writeToLog(output);
+}
+
+function sessionEnded (session, reason) {
+  let output = `${_getLogHeader('terminated')}
```
Comment (davidmorton-gen) [inline test/test-pages/gcba/sdk-controller.ts:291]: Codacy has a fix for the issue: ['output' is never reassigned. Use 'const' instead.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9580329)
```suggestion
  const output = `${_getLogHeader('terminated')}
```

File: test/unit/client.test.ts [hunk 3]
```diff
@@ -1113,4 +1117,45 @@ describe('Client', () => {
...
+    
+    it('should init with jwt', async () => {
+      jwtDecodeSpy.mockReturnValue({
+        name: 'scooby',
+        org: 'myorg',
+        data: {
+          uid: 'myuserid',
+          jid: 'myjid',
+        }
+      } as any);
```
Comment (davidmorton-gen) [inline test/unit/client.test.ts:1147]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9580329)
```suggestion
      } as unknown);
```

File: test/test-pages/gcba/sdk-controller.ts [hunk 4]
```diff
@@ -0,0 +1,307 @@
...
+  webrtcSdk.rejectPendingSession({ conversationId });
+}
+
+function endSession ({ conversationId }) {
+  webrtcSdk.endSession({ conversationId });
+}
+
+function exposeGlobalFunctions () {
+  (window as any).acceptPendingSession = acceptPendingSession;
+  (window as any).rejectPendingSession = rejectPendingSession;
```
Comment (davidmorton-gen) [inline test/test-pages/gcba/sdk-controller.ts:71]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9580329)
```suggestion
  (window as unknown).rejectPendingSession = rejectPendingSession;
```

File: test/test-pages/gcba/platform-client.ts [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+    client.setEnvironment((window as any).environments[environment].uri);
+  }
+
+  function setAuthTextVisible (visible) {
+    const style = visible ? 'visible' : 'hidden';
+    document.getElementById('auth-text').style.visibility = style;
+  }
+
+  function signout () {
+    const environment: any = getInputValue('environment');
```
Comment (davidmorton-gen) [inline test/test-pages/gcba/platform-client.ts:17]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9580329)
```suggestion
    const environment: unknown = getInputValue('environment');
```

File: test/test-pages/gcba/sdk-controller.ts [hunk 6]
```diff
@@ -0,0 +1,307 @@
...
+function ready () {
+  if (!webrtcSdk._personDetails) {
+    webrtcSdk.startScreenShare();
+  }
+  writeToLog('webrtcSDK ready event emitted');
+}
+
+// pendingSession - {id, address, conversationId, autoAnswer}
+function pendingSession (options) {
+  let output = `${_getLogHeader('pendingSession')}
```
Comment (davidmorton-gen) [inline test/test-pages/gcba/sdk-controller.ts:102]: Codacy has a fix for the issue: ['output' is never reassigned. Use 'const' instead.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9580329)
```suggestion
  const output = `${_getLogHeader('pendingSession')}
```

File: test/test-pages/gcba/app-controller.ts [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+  if (isHidden) {
+    element.classList.remove(displayNone);
+  } else {
+    element.classList.add(displayNone);
+  }
+}
+
+export function initialize () {
+  const envInput = getInputValue('environment');
+  const environmentInfo = (window as any).environments[envInput];
```
Comment (davidmorton-gen) [inline test/test-pages/gcba/app-controller.ts:38]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9580329)
```suggestion
  const environmentInfo = (window as unknown).environments[envInput];
```

File: test/test-pages/gcba/sdk-controller.ts [hunk 8]
```diff
@@ -0,0 +1,307 @@
...
+    sessionId: ${params.sessionId}
+    conversationId: ${params.conversationId}`;
+
+  pendingSessions = pendingSessions.filter(s => s.conversationId !== params.conversationId);
+  renderPendingSessions();
+  writeToLog(output);
+}
+
+function handledPendingSession (params) {
+  let output = `${_getLogHeader('handledPendingSession')}
```
Comment (davidmorton-gen) [inline test/test-pages/gcba/sdk-controller.ts:234]: Codacy has a fix for the issue: ['output' is never reassigned. Use 'const' instead.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9580329)
```suggestion
  const output = `${_getLogHeader('handledPendingSession')}
```

File: src/client.ts [hunk 9]
```diff
@@ -128,6 +134,14 @@ export class GenesysCloudWebrtcSdk extends (EventEmitter as { new(): StrictEvent
     /* grab copies or valid objects */
     const defaultsOptions = options.defaults || {};
 
+    let allowedSessionTypes = options.allowedSessionTypes || [SessionTypes.softphone, SessionTypes.collaborateVideo, SessionTypes.acdScreenShare];
```
Comment (djhouseknecht) [inline src/client.ts:137]: NIT: this could probably be a `const`

File: src/client.ts [hunk 10]
```diff
@@ -510,6 +526,23 @@ export class GenesysCloudWebrtcSdk extends (EventEmitter as { new(): StrictEvent
       });
   }
 
+  parseJwt (): void {
+    const decoded: JWTDetails = jwtDecode(this._config.jwt);
```
Comment (djhouseknecht) [inline src/client.ts:530]: Does this throw an error if the jwt is malformed? If yes, maybe we should emit on `sdkError`. Although, if we through this in the constructor, there won't be any listeners yet.

File: src/types/interfaces.ts [hunk 11]
```diff
@@ -52,6 +52,12 @@ export interface ISdkConfig {
    */
   accessToken?: string;
 
+  /**
+   * jwt providing limited functionality. For the time being,
+   * this is limited to agent screen recording functionality.
+   */
+  jwt?: string;
```
Comment (djhouseknecht) [inline src/types/interfaces.ts:59]: We will want to add this to the docs.

**Review by djhouseknecht — COMMENTED** [review]
Do we want to commit all the `test/test-pages/dev**` built files?

File: src/client.ts (hunk 9 above)
Comment (jensengar) [reply src/client.ts:137]: This can't be a const because it gets reassigned a few lines down if we are using jwt.

File: src/client.ts (hunk 10 above)
Comment (jensengar) [reply src/client.ts:530]: I'll check this.

File: test/test-pages/gcba/sdk-controller.ts (hunk 8 above)
Comment (davidmorton-gen) [inline test/test-pages/gcba/sdk-controller.ts:234]: Codacy has a fix for the issue: ['output' is never reassigned. Use 'const' instead.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9580329)
```suggestion
  const output = `${_getLogHeader('handledPendingSession')}
```

File: test/test-pages/gcba/sdk-controller.ts [hunk 12]
```diff
@@ -0,0 +1,307 @@
...
+    <td class="${isSessionActive ? 'text-success' : 'text-danger'}">
+      ${session.state}
+    </td>
+  </tr>`
+  });
+  tableBody.innerHTML = html;
+}
+
+function cancelPendingSession (params) {
+  let output = `${_getLogHeader('cancelPendingSession')}
```
Comment (davidmorton-gen) [inline test/test-pages/gcba/sdk-controller.ts:224]: Codacy has a fix for the issue: ['output' is never reassigned. Use 'const' instead.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9580329)
```suggestion
  const output = `${_getLogHeader('cancelPendingSession')}
```

File: test/test-pages/gcba/platform-client.ts (hunk 1 above)
Comment (davidmorton-gen) [inline test/test-pages/gcba/platform-client.ts:45]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9580329)
```suggestion
        (window as unknown).conversationsAPI = new (window as any).platformClientConversationsApi();
```

File: test/test-pages/gcba/sdk-controller.ts [hunk 13]
```diff
@@ -0,0 +1,307 @@
+/* global MediaStream */
+import GenesysCloudWebrtSdk, { ConversationUpdate, IExtendedMediaSession, ISdkConfig, ISdkConversationUpdateEvent, IStoredConversationState, SdkError, SessionTypes } from '../../../src';
+import { getInputValue, writeToLog } from './utils';
+
+let webrtcSdk: GenesysCloudWebrtSdk;
+let pendingSessions = [];
+
+let conversationUpdatesToRender: {[conversationId: string]: IExtendedMediaSession} = { };
```
Comment (davidmorton-gen) [inline test/test-pages/gcba/sdk-controller.ts:8]: Codacy has a fix for the issue: ['conversationUpdatesToRender' is never reassigned. Use 'const' instead.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9580329)
```suggestion
const conversationUpdatesToRender: {[conversationId: string]: IExtendedMediaSession} = { };
```

File: test/test-pages/gcba/sdk-controller.ts [hunk 14]
```diff
@@ -0,0 +1,307 @@
...
+}
+
+function endSession ({ conversationId }) {
+  webrtcSdk.endSession({ conversationId });
+}
+
+function exposeGlobalFunctions () {
+  (window as any).acceptPendingSession = acceptPendingSession;
+  (window as any).rejectPendingSession = rejectPendingSession;
+  (window as any).endSession = endSession;
```
Comment (davidmorton-gen) [inline test/test-pages/gcba/sdk-controller.ts:72]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9580329)
```suggestion
  (window as unknown).endSession = endSession;
```

File: test/test-pages/gcba/platform-client.ts [hunk 15]
```diff
@@ -0,0 +1,170 @@
...
+  }
+
+  async function fetchNewJwt () {
+    const accessToken = manualAuthInput.value;
+
+    if (!accessToken) {
+      throw new Error('Must have an access token to');
+    }
+
+    const url = `https://api.${(window as any).environments[getInputValue('environment')].uri}/api/v2/screenrecording/token`;
```
Comment (davidmorton-gen) [inline test/test-pages/gcba/platform-client.ts:120]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9580329)
```suggestion
    const url = `https://api.${(window as unknown).environments[getInputValue('environment')].uri}/api/v2/screenrecording/token`;
```

File: test/test-pages/gcba/platform-client.ts [hunk 16]
```diff
@@ -0,0 +1,170 @@
...
+      });
+  }
+
+  function authenticateFromToken (token) {
+    const environment = getInputValue('environment');
+    initClientEnvironment(environment);
+
+    window.localStorage.setItem(`${persitentName}_auth_data`, JSON.stringify({ accessToken: token }));
+    client.authentications['PureCloud OAuth'].accessToken = token;
+    (window as any).conversationsAPI = new (window as any).platformClient.ConversationsApi();
```
Comment (davidmorton-gen) [inline test/test-pages/gcba/platform-client.ts:59]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9580329)
```suggestion
    (window as unknown).conversationsAPI = new (window as any).platformClient.ConversationsApi();
```

File: test/test-pages/gcba/platform-client.ts (hunk 5 above)
Comment (davidmorton-gen) [inline test/test-pages/gcba/platform-client.ts:17]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9580329)
```suggestion
    const environment: unknown = getInputValue('environment');
```

**Review by zservies — APPROVED** [review]

File: test/test-pages/gcba/sdk-controller.ts (hunk 2 above)
Comment (davidmorton-gen) [inline test/test-pages/gcba/sdk-controller.ts:291]: Codacy has a fix for the issue: ['output' is never reassigned. Use 'const' instead.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9580329)
```suggestion
  const output = `${_getLogHeader('terminated')}
```

File: test/unit/client-private.test.ts [hunk 17]
```diff
@@ -2,15 +2,59 @@ import { GenesysCloudWebrtcSdk } from '../../src/client';
...
+      chat: {
+        jabberId: 'myJid'
+      }
+    };
+
+    let connectedCb: () => void;
+    const connectSpy = jest.fn().mockImplementation(() => {
+      connectedCb();
+    });
+    (StreamingClient as any).mockReturnValue({
```
Comment (davidmorton-gen) [inline test/unit/client-private.test.ts:40]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9580329)
```suggestion
    (StreamingClient as unknown).mockReturnValue({
```

File: test/test-pages/gcba/sdk-controller.ts (hunk 14 above)
Comment (davidmorton-gen) [inline test/test-pages/gcba/sdk-controller.ts:72]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9580329)
```suggestion
  (window as unknown).endSession = endSession;
```

File: test/test-pages/gcba/platform-client.ts (hunk 15 above)
Comment (davidmorton-gen) [inline test/test-pages/gcba/platform-client.ts:120]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9580329)
```suggestion
    const url = `https://api.${(window as unknown).environments[getInputValue('environment')].uri}/api/v2/screenrecording/token`;
```

File: test/test-pages/gcba/sdk-controller.ts (hunk 4 above)
Comment (davidmorton-gen) [inline test/test-pages/gcba/sdk-controller.ts:71]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9580329)
```suggestion
  (window as unknown).rejectPendingSession = rejectPendingSession;
```

File: test/test-pages/gcba/sdk-controller.ts [hunk 18]
```diff
@@ -0,0 +1,307 @@
...
+  output += `  level: ${level}\n  message: ${message}\n  details: ${details}`;
+
+  const logTraces = (document.getElementById('log-traces-check') as any).checked;
+  if (logTraces) {
+    writeToLog(output);
+  }
+}
+
+function error (error: SdkError) {
+  let output = `${_getLogHeader('error')}
```
Comment (davidmorton-gen) [inline test/test-pages/gcba/sdk-controller.ts:284]: Codacy has a fix for the issue: ['output' is never reassigned. Use 'const' instead.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9580329)
```suggestion
  const output = `${_getLogHeader('error')}
```

File: test/test-pages/gcba/sdk-controller.ts [hunk 19]
```diff
@@ -0,0 +1,307 @@
...
+  ];
+
+  webrtcSdk.acceptSession({ conversationId: session.conversationId, sessionType: session.sessionType, mediaStream: screenStream, screenRecordingMetadatas });
+}
+
+function trace (level, message, details) {
+  let output = `${_getLogHeader('trace')}\n`;
+  output += `  level: ${level}\n  message: ${message}\n  details: ${details}`;
+
+  const logTraces = (document.getElementById('log-traces-check') as any).checked;
```
Comment (davidmorton-gen) [inline test/test-pages/gcba/sdk-controller.ts:277]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9580329)
```suggestion
  const logTraces = (document.getElementById('log-traces-check') as unknown).checked;
```

File: test/test-pages/gcba/platform-client.ts [hunk 20]
```diff
@@ -0,0 +1,170 @@
...
+
+  function authenticate () {
+    const environment = getInputValue('environment');
+    const clientId = (window as any).environments[environment].clientId;
+    initClientEnvironment(environment);
+
+    client.loginImplicitGrant(clientId, window.location.href)
+      .then(() => {
+        const authInfo = JSON.parse(window.localStorage.getItem(`${persitentName}_auth_data`));
+        (client as any).authentications['PureCloud Auth'].accessToken = authInfo.accessToken;
```
Comment (davidmorton-gen) [inline test/test-pages/gcba/platform-client.ts:43]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9580329)
```suggestion
        (client as unknown).authentications['PureCloud Auth'].accessToken = authInfo.accessToken;
```

File: test/test-pages/gcba/sdk-controller.ts [hunk 21]
```diff
@@ -0,0 +1,307 @@
...
+    sessionId: ${params.sessionId}
+    conversationId: ${params.conversationId}`;
+
+  pendingSessions = pendingSessions.filter(s => s.conversationId !== params.conversationId);
+  renderPendingSessions();
+  writeToLog(output);
+}
+
+async function sessionStarted (session: IExtendedMediaSession) {
+  let output = `${_getLogHeader('sessionStarted')}
```
Comment (davidmorton-gen) [inline test/test-pages/gcba/sdk-controller.ts:244]: Codacy has a fix for the issue: ['output' is never reassigned. Use 'const' instead.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9580329)
```suggestion
  const output = `${_getLogHeader('sessionStarted')}
```

File: test/unit/client.test.ts [hunk 22]
```diff
@@ -1113,4 +1117,61 @@ describe('Client', () => {
...
```
Comment (davidmorton-gen) [inline test/unit/client.test.ts:1147]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9580329)
```suggestion
      } as unknown);
```

File: test/test-pages/gcba/platform-client.ts (hunk 1 above)
Comment (davidmorton-gen) [inline test/test-pages/gcba/platform-client.ts:45]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9580329)
```suggestion
        (window as unknown).conversationsAPI = new (window as any).platformClientConversationsApi();
```

File: test/test-pages/gcba/platform-client.ts (hunk 5 above)
Comment (davidmorton-gen) [inline test/test-pages/gcba/platform-client.ts:17]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9580329)
```suggestion
    const environment: unknown = getInputValue('environment');
```

File: test/test-pages/gcba/sdk-controller.ts [hunk 23]
```diff
@@ -0,0 +1,307 @@
...
+  options.logLevel = 'info';
+  // for sumo debugging
+  // options.optOutOfTelemetry = true;
+
+  const SDK = GenesysCloudWebrtSdk;
+  (window as any).SDK = SDK;
+
+  webrtcSdk = new SDK(options);
+  (window as any).webrtcSdk = webrtcSdk;
+  (window as any).sdk = webrtcSdk;
```
Comment (davidmorton-gen) [inline test/test-pages/gcba/sdk-controller.ts:34]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9580329)
```suggestion
  (window as unknown).sdk = webrtcSdk;
```

File: test/test-pages/gcba/sdk-controller.ts (hunk 12 above)
Comment (davidmorton-gen) [inline test/test-pages/gcba/sdk-controller.ts:224]: Codacy has a fix for the issue: ['output' is never reassigned. Use 'const' instead.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9580329)
```suggestion
  const output = `${_getLogHeader('cancelPendingSession')}
```

File: test/test-pages/gcba/sdk-controller.ts [hunk 24]
```diff
@@ -0,0 +1,307 @@
...
+  options.environment = environmentData.uri;
+  options.logLevel = 'info';
+  // for sumo debugging
+  // options.optOutOfTelemetry = true;
+
+  const SDK = GenesysCloudWebrtSdk;
+  (window as any).SDK = SDK;
+
+  webrtcSdk = new SDK(options);
+  (window as any).webrtcSdk = webrtcSdk;
```
Comment (davidmorton-gen) [inline test/test-pages/gcba/sdk-controller.ts:33]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9580329)
```suggestion
  (window as unknown).webrtcSdk = webrtcSdk;
```

File: test/test-pages/gcba/platform-client.ts (hunk 16 above)
Comment (davidmorton-gen) [inline test/test-pages/gcba/platform-client.ts:59]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9580329)
```suggestion
    (window as unknown).conversationsAPI = new (window as any).platformClient.ConversationsApi();
```

File: test/test-pages/gcba/platform-client.ts [hunk 25]
```diff
@@ -0,0 +1,170 @@
+import { getCurrentUrlParams, getInputValue } from './utils';
+const client = (window as any).platformClient.ApiClient.instance;
+const persitentName = 'sdk_test';
+
+(function () {
+  function initClientEnvironment (environment) {
+    client.setPersistSettings(true, persitentName);
+    client.setEnvironment((window as any).environments[environment].uri);
```
Comment (davidmorton-gen) [inline test/test-pages/gcba/platform-client.ts:8]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9580329)
```suggestion
    client.setEnvironment((window as unknown).environments[environment].uri);
```

File: test/test-pages/gcba/platform-client.ts [hunk 26]
```diff
@@ -0,0 +1,170 @@
+import { getCurrentUrlParams, getInputValue } from './utils';
+const client = (window as any).platformClient.ApiClient.instance;
```
Comment (davidmorton-gen) [inline test/test-pages/gcba/platform-client.ts:2]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9580329)
```suggestion
const client = (window as unknown).platformClient.ApiClient.instance;
```

**Pull request merged**